diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index 2838c3d..47beee6 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -873,7 +873,7 @@ sub vms_info {
         # systems are perfectly capable of executing binaries targeting
         # Froyo. Keep in mind that in the nutshell Android builds are
         # about JNI, i.e. shared libraries, not applications.
-        cflags           => add(picker(default => "-mandroid -fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
+        cflags           => add(picker(default => "-fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
         bin_cflags       => "-pie",
     },
     "android-x86" => {
@@ -913,7 +913,7 @@ sub vms_info {
 
     "android64" => {
         inherit_from     => [ "linux-generic64" ],
-        cflags           => add(picker(default => "-mandroid -fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
+        cflags           => add(picker(default => "-fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
         bin_cflags       => "-pie",
     },
     "android64-aarch64" => {
