diff --git a/crypto/chacha/asm/chacha-armv4.pl b/crypto/chacha/asm/chacha-armv4.pl
index b5e21e4..2e82f28 100755
--- a/crypto/chacha/asm/chacha-armv4.pl
+++ b/crypto/chacha/asm/chacha-armv4.pl
@@ -179,7 +179,7 @@ $code.=<<___;
 .code	32
 #endif
 
-#if defined(__thumb2__) || defined(__clang__)
+#if defined(__thumb2__)
 #define ldrhsb	ldrbhs
 #endif

diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index f32ccbb..d23e78a 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -919,6 +919,7 @@ sub vms_info {
     #
     "android-armeabi" => {
         inherit_from     => [ "android", asm("armv4_asm") ],
+        ex_libs          => add("-latomic"),
     },
     "android-mips" => {
         inherit_from     => [ "android", asm("mips32_asm") ],
